tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/bugs/ProjectImporter.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2018 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.tests.bugs;
+
+import static com.google.common.collect.Sets.newHashSet;
+import static org.eclipse.n4js.tests.util.ProjectTestsUtils.importProject;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.PrintWriter;
+
+import org.eclipse.n4js.utils.io.FileUtils;
+import org.eclipse.xtext.ui.testing.util.IResourcesSetupUtil;
+
+/**
+ * Simple project importer implementation.
+ */
+class ProjectImporter {
+
+	/**
+	 * The NOOP importer. Does not import anything into the workspace.
+	 */
+	public static ProjectImporter NOOP = new ProjectImporter();
+
+	private final File rootFolder;
+
+	private ProjectImporter() {
+		this(FileUtils.createTempDirectory().toFile());
+	}
+
+	/**
+	 * Creates a project importer with the root folder of all projects that has to be imported for the test.
+	 *
+	 * @param rootFolder
+	 *            the root folder.
+	 */
+	public ProjectImporter(final File rootFolder) {
+		assertNotNull("Root folder cannot be null.", rootFolder);
+		assertTrue("Root folder does not exist: " + rootFolder, rootFolder.exists());
+		assertTrue("Root folder must be a folder. But was a file: " + rootFolder, rootFolder.isDirectory());
+		assertTrue("Root folder directory content cannot be read: " + rootFolder, rootFolder.canRead());
+		assertTrue("No files were found in the directory: " + rootFolder, null != rootFolder.listFiles());
+		this.rootFolder = rootFolder;
+	}
+
+	void importProjects() throws Exception {
+		for (final File file : rootFolder.listFiles()) {
+			if (file.exists() && file.isDirectory() && null != file.listFiles() && 0 < file.listFiles().length) {
+
+				if (!newHashSet(file.list()).contains("_project")) {
+					final String projectName = file.getName();
+					AbstractIDEBUG_Test.LOGGER.info("Project \'" + projectName
```

Comment:
What kind of logging concept is that? Does it mean that the ProjectImporter should only used by ``AbstractIDEBUG_Test``s?

Reply:
I guess refactoring just kept the reference. I will change it